src/mac/safari-extension.mm

```diff
@@ -71,13 +68,18 @@ - (void)beginRequestWithExtensionContext:(NSExtensionContext *)context
     NSLog(@"web-eid-safari-extension: web-eid-safari isRunning: %d", isRunning);
     if (!isRunning) {
         if (NSURL *appURL = [NSWorkspace.sharedWorkspace URLForApplicationWithBundleIdentifier:WebEidApp]) {
+            setValue(WebEidStarting, @(true));
             if ([NSWorkspace.sharedWorkspace launchApplication:appURL.path]) {
                 NSLog(@"web-eid-safari-extension: started app");
+                do {
+                    [NSThread sleepForTimeInterval:1.0f];
+                    NSLog(@"web-eid-safari-extension: waiting to be running %@", [getUserDefaults() objectForKey:WebEidStarting]);
+                } while ([getUserDefaults() boolForKey:WebEidStarting]);
```

Comment:
Great, so it worked out as we discussed! However, we need to add a timeout here, otherwise this may result in an infinite loop in case of errors in `main()` above so that `takeValue(WebEidStarting)` is either not reached or fails.

Reply:
Added timout